src/main/java/com/android/volley/RequestQueue.java

```diff
@@ -38,11 +38,29 @@
     // TODO: This should not be a generic class, because the request type can't be determined at
     // compile time, so all calls to onRequestFinished are unsafe. However, changing this would be
     // an API-breaking change. See also: https://github.com/google/volley/pull/109
+    @Deprecated // Use RequestEventListener instead.
     public interface RequestFinishedListener<T> {
         /** Called when a request has finished processing. */
         void onRequestFinished(Request<T> request);
     }
 
+    /** Request event types listeners will be notified about. */
+    public enum RequestEvent {
```

Comment:
I'm on the fence about this, but feel slightly that this should be a set of @IntDef'd constants rather than an enum given the Android team's long-standing recommendation against enums and the fact that we're a commonly-used library and can't control the optimization processes that apps using it might have configured.

Reply:
Done